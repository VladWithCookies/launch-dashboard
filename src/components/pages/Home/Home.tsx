import { useMemo } from 'react';
import { formatISO, add } from 'date-fns';
import { pipe, map, propOr } from 'ramda';
import { useQuery } from '@tanstack/react-query';
import { Center, Spinner } from '@chakra-ui/react';

import { ENDPOINTS } from 'constants/api';
import { getLaunches } from 'api';
import Map from 'components/organisms/Map';
import MainLayout from 'components/templates/MainLayout';

export default function Home() {
  const startDate = formatISO(new Date());
  const endDate = formatISO(add(new Date(), { weeks: 3 }));

  const { isLoading, data } = useQuery(
    [ENDPOINTS.LAUNCH, startDate, endDate],
    () => getLaunches({ startDate, endDate })
  );

  const markers = useMemo(() => pipe<[Data], Result[], Marker[]>(
    propOr([], 'results'),
    map(({ id, name, pad: { latitude, longitude } }: Result) => ({
      id,
      name,
      coordinates: [Number(longitude), Number(latitude)],
    }))
  )(data), [data]);

  return (
    <MainLayout>
      <Center h="400px">
        {isLoading ? <Spinner size="xl" /> : <Map markers={markers} />}
      </Center>
    </MainLayout>
  );
}
