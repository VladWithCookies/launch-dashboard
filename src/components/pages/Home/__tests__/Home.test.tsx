import { render, screen, waitFor } from 'utils/tests/renderer';
import Home from '../Home';

describe('Home', () => {
  it('matches snapshot', async () => {
    const { container } = render(<Home />);

    await waitFor(() => {
      expect(screen.getByTestId('map')).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
