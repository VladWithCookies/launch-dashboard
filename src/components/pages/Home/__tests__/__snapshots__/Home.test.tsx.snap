// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<div>
  <div
    class="chakra-container css-0"
  >
    <div
      class="chakra-stack css-0"
    >
      <div
        class="css-0"
      >
        <h1
          class="chakra-heading css-0"
        >
          Moonshot Calendar Inc.
        </h1>
      </div>
      <div
        class="css-0"
      >
        <svg
          class="rsm-svg "
          data-testid="map"
          viewBox="0 0 800 600"
        >
          <g
            class="rsm-geographies "
          />
          <g
            class="rsm-marker "
            transform="translate(NaN, NaN)"
          >
            <circle
              fill="#F00"
              r="10"
              stroke="#fff"
              stroke-width="2"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
